github_slack_bot/main.py

```diff
@@ -0,0 +1,14 @@
+import os
+from utils import slack
+from utils import git
+
+if __name__ == "__main__":
+    # TODO: add config system to grab values from a standard chain (e.g. cli params, env vars, .yml file)
+    slack_api_token = os.environ.get('SLACK_API_TOKEN')
+    github_repo = os.environ.get('GITHUB_REPO')
+    github_api_token = os.environ.get('GITHUB_REPO')
+
+    git_client = git.GitClient()
+    slack_bot = slack.SlackBot(slack_api_token, git_client)
```

Comment:
I'm curious what'll happen if any of the functions in "main" raise an exception. Any way to handle them and make the program exit gracefully?

Reply:
I improved the logging and exception handling in the main() function so that it gives more useful information before shutting down the bot. I also made it handle Ctrl+C from users.